### DISCUSSION TITLE
Installing Paradox Launcher for Cities: Skylines II

### DISCUSSION THREAD

**Discussion by chrisvander** [Whisky-App/Whisky#556] [Q&A] [answered]
Hi, attempting to get CS:II up and running on my M1 Max MBP. The Paradox launcher won't install - getting errors in the launcher logs like:
```
The `launcherpath` file does not exists:
C:\users\crossover\AppData\Local\Paradox Interactive\launcherpath
2023-10-24 16:17:01 [main] [INFO] Trying to fix existing installation in "/passive" mode
2023-10-24 16:17:01 [main] [ERROR] Fixing existing installation failed:
exit status 1605
```

In the launcher-installer.log file, I see:

```
SFXCA: Extracting custom action to temporary directory: C:\users\crossover\Temp\msi5ef2.tmp-\
SFXCA: Failed to get requested CLR info. Error code 0x80004005
SFXCA: Ensure that the proper version of the .NET Framework is installed, or that there is a matching supportedRuntime element in CustomAction.config. If you are binding to .NET 4 or greater add useLegacyV2RuntimeActivationPolicy=true to the <startup> element.
Action ended 4:23:41: InstallAdditions. Return value 0.
Action ended 4:23:41: InstallFinalize. Return value 0.
Action ended 4:23:41: INSTALL. Return value 0.
```

There's another potential error but the .NET error seems more likely:
```
DEBUG: Error 2726:  Action not found: MigrateFeatureStates
The installer has encountered an unexpected error installing this package. This may indicate a problem with this package. The error code is 2726. 
```

How do I install .NET into the bottle correctly?

**Comment by IsaacMarovitz** [comment]
I refunded the game. I don't have a solution for now.

**Comment by IsaacMarovitz** [comment, marked as answer]
Installing `dotnet48` with Winetricks fixes the installer.

**Reply by chrisvander** [reply]
Wicked. `dotnet48` installed perfectly for me. Time to try setup again after rage deleting it all...

**Reply by IsaacMarovitz** [reply]
Better than rage refunding lol

**Reply by TheMeinerLP** [reply]
I still get in the logs `FXCA: Failed to load mscoree.dll (Error code 126). This custom action requires the .NET Framework to be installed.`

**Reply by Junyi-99** [reply]
Have you successfully installed `dotnet48` using `winetricks` ?

**Reply by mmmeff** [reply]
Could anyone please be so kind as to point me to how one should use winetricks with a whisky bottle?

**Reply by MPFGlaser** [reply]
@mmmeff There should be a Winetricks... button on the bottom right if you have a bottle selected.
<img width="1012" alt="image" src="https://github.com/Whisky-App/Whisky/assets/31521977/d6936650-72e8-4817-8cff-30c7f4b747a5">
Clicking that and filling in a command such as `dotnet48` should do the trick. Don't add any other things in there, like the Wine wiki might suggest. That's to get Winetricks working, which is something the button already takes care of.

**Comment by chrisvander** [comment]
Made it past launcher but getting a Unity error now. Saw something online about volumetrics. I'm going to figure it out another time, but seems like people are having a decent experience on hardware that isn't that powerful, so I remain hopeful in my M1 Max MBP...

**Comment by Junyi-99** [comment]
Hey I cannot install dotnet 48 using `winetricks`.

Can you help me look at the problem? Here is the log:

```log
(base) ➜  ~ PATH="/Users/junyi/Library/Application Support/com.isaacmarovitz.Whisky/Libraries/Wine/bin:$PATH" WINE=wine64 WINEPREFIX="/Users/junyi/Library/Containers/com.isaacmarovitz.Whisky/Bottles/9D98FCBF-55DC-4455-960F-364381A0E61D" "/Users/junyi/Library/Application Support/com.isaacmarovitz.Whisky/Libraries/winetricks" dotnet48
warning: taskset/cpuset not available on your platform!
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
------------------------------------------------------
WINEPREFIX INFO:
Drive C: .
..
.DS_Store
Program Files
Program Files (x86)
ProgramData
users
windows

Registry info:
/Users/junyi/Library/Containers/com.isaacmarovitz.Whisky/Bottles/9D98FCBF-55DC-4455-960F-364381A0E61D/system.reg:#arch=win64
/Users/junyi/Library/Containers/com.isaacmarovitz.Whisky/Bottles/9D98FCBF-55DC-4455-960F-364381A0E61D/user.reg:#arch=win64
/Users/junyi/Library/Containers/com.isaacmarovitz.Whisky/Bottles/9D98FCBF-55DC-4455-960F-364381A0E61D/userdef.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
warning: wine64 cmd.exe /c echo '%AppData%' returned empty string, error message "" 
------------------------------------------------------
(base) ➜  ~ 

```

**Reply by MPFGlaser** [reply]
What command did you try exactly? I also faced some troubles myself, until I did the following:  
1. Select your bottle in Whisky
2. Click the "Winetricks..." button in the bottom right.
3. Enter `dotnet48 win10` and click "Run"
4. Follow the on-screen prompts

You might need to kill your bottles using `File > Kill All Bottles` in the top bar when having Whisky on the foreground.

**Reply by Junyi-99** [reply]
Yes, your solution is correct. (Especially to kill all bottles first.)

However, after installing `dotnet48`, I met "Unity Crash" problem...

**Reply by MPFGlaser** [reply]
Also facing a crash here. Probably the same thing. Trying to figure out what causes it (and if it’s fixable). I’ll report back if and when I’ve found something.

**Reply by Junyi-99** [reply]
Same.

**Reply by chrisvander** [reply]
Yup, also experiencing a Unity crash. I seem to remember hearing something was related to volumetrics. I wonder if the configuration can be changed before launch to disable volumetrics - maybe it wouldn't crash?

**Reply by MPFGlaser** [reply]
I tried this myself already by copying a configuration from my Windows PC to my Mac with those settings turned off. Unfortunately, it still crashes the game upon boot. I hope a fix will be found by someone soon.

**Reply by chrisvander** [reply]
Damn. Worth a shot.

**Reply by Junyi-99** [reply]
I suggest we go to https://github.com/Whisky-App/Whisky/issues/573 to discuss ways to get the game running.

**Comment by Junyi-99** [comment]
I suggest we go to #573 to discuss ways to get the game running.

**Comment by eigenein** [comment]
I'm afraid I'm getting another error, even with `dotnet48` installed, ignoring does not help it:

```
SFXCA: Extracting custom action to temporary directory: C:\users\crossover\Temp\msi58ec.tmp-\
SFXCA: Binding to CLR version v4.0.30319
SFXCA: RUNDLL32 returned error code: -1073740286
```

- Whisky 2.2.1 (34)
- Paradox Launcher Installer 2023_16

<img width="443" alt="mscorlib recursive resource lookup bug"  title="mscorlib recursive resource lookup bug" src="https://github.com/Whisky-App/Whisky/assets/308906/fea4aee0-92c5-47af-9593-c543b0cda2a4">

**Reply by andylmfao** [reply]
Same issue here. Tried everything. Did you ever get it to work?
